src/bin/josh-filter.rs

```diff
@@ -113,11 +124,53 @@ fn run_filter(args: Vec<String>) -> josh::JoshResult<i32> {
         josh::apply_filter_to_refs(
             &repo,
             &*viewobj,
-            &[(src.clone(), t)],
+            &[(src.clone(), t.clone())],
             &mut fm,
             &mut backward_maps.write().unwrap(),
         );
 
+        let mut all_dirs = vec![];
+
+        if check_permissions {
+            let result_tree = repo.find_reference(&t)?.peel_to_tree()?;
+
+            result_tree.walk(git2::TreeWalkMode::PreOrder, |_, entry| {
+                let name = entry.name().unwrap();
+                if name.starts_with("JOSH_ORIG_PATH_") {
+                    let dirname = format!(
+                        "{}",
+                        josh::from_ns(&name.replacen("JOSH_ORIG_PATH_", "", 1))
+                    );
+                    all_dirs.push(dirname);
+                }
+                git2::TreeWalkResult::Ok
+            })?;
+        }
+
+        let mut dedup = vec![];
+
+        for w in all_dirs.as_slice().windows(2) {
+            if let [a, b, ..] = w {
+                if !b.starts_with(a) {
+                    dedup.push(a.to_owned());
+                }
+            }
+        }
+
+        if let Some(cp) = args.value_of("check-permission") {
+            let permission_regex = regex::Regex::new(cp)?;
+
+            let mut allowed = dedup.len() != 0;
```

Comment:
if dedup is empty, why even do all of this?

Reply:
Do what? If dedup is empty it only prints allowed = true and does nothing else.